solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClientJdkImpl.java

```diff
@@ -0,0 +1,443 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.client.solrj.impl;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.net.CookieStore;
+import java.net.InetSocketAddress;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpTimeoutException;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.net.ssl.SSLContext;
+import org.apache.solr.client.solrj.ResponseParser;
+import org.apache.solr.client.solrj.SolrRequest;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.request.RequestWriter;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.apache.solr.common.util.ContentStream;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.common.util.ObjectReleaseTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpSolrClientJdkImpl extends Http2SolrClientBase {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private static final String USER_AGENT =
+      "Solr[" + MethodHandles.lookup().lookupClass().getName() + "] 1.0";
+
+  private HttpClient client;
+
+  protected HttpSolrClientJdkImpl(String serverBaseUrl, HttpSolrClientJdkImpl.Builder builder) {
+    super(serverBaseUrl, builder);
+
+    HttpClient.Redirect followRedirects =
+        Boolean.TRUE.equals(builder.followRedirects)
+            ? HttpClient.Redirect.NORMAL
+            : HttpClient.Redirect.NEVER;
+    HttpClient.Builder b = HttpClient.newBuilder().followRedirects(followRedirects);
+    if (builder.sslContext != null) {
+      b.sslContext(builder.sslContext);
+    }
+    if (builder.executor != null) {
+      b.executor(builder.executor);
+    }
+    if (builder.proxyHost != null) {
+      if (builder.proxyIsSocks4) {
+        log.warn(
+            "Socks4 is likely not supported by this client.  See https://bugs.openjdk.org/browse/JDK-8214516");
```

Comment:
I was on the fence with this one.  I chose the log warning over the exception because (1) `Http2SolrClient` already does the same thing for `keyStoreReloadIntervalSecs`.  and (2) JDK-8214516 _could_ maybe get resolved in such a way that a future JDK could work with socks2 without any changes on our end.   But then again, looking at the last (5 year old) comment on that ticket, this doesn't seem likely!

Reply:
You've thought about it more than me; I defer to your judgement.